jena-base/src/main/java/org/apache/jena/atlas/lib/cache/CacheCaffeine.java

```diff
@@ -44,12 +44,18 @@ public CacheCaffeine(int size) {
     }
 
     public CacheCaffeine(int size, BiConsumer<K, V> dropHandler) {
+        this(size, dropHandler, -1);
+    }
+
+    public CacheCaffeine(int size, BiConsumer<K, V> dropHandler, double initialCapacityFactor) {
         @SuppressWarnings("unchecked")
         Caffeine<K,V> builder = (Caffeine<K,V>)Caffeine.newBuilder()
-            .maximumSize(size)
-            .initialCapacity(size/4)
-            // Eviction immediately using the caller thread.
-            .executor(c->c.run());
+            .maximumSize(size);
+        if (initialCapacityFactor > 0.0) {
```

Comment:
Also needs an upper bound of `1.0` here otherwise you could try and set the initial capacity higher than the maximum size

Reply:
@rvesse thanks for catching and reporting this! I just pushed a fix with the correct bounds.